src/python/pants/build_graph/import_remote_sources_mixin.py

```diff
@@ -0,0 +1,124 @@
+# coding=utf-8
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from builtins import str
+
+from future.utils import string_types
+
+from pants.build_graph.address_lookup_error import AddressLookupError
+from pants.build_graph.target import Target
+from pants.util.memo import memoized_property
+from pants.util.meta import AbstractClass, classproperty
+from pants.util.objects import TypeConstraintError
+
+
+class ImportRemoteSourcesMixin(Target, AbstractClass):
+  """A Target Mixin to be used when a target declares another target type to be imported."""
```

Comment:
So, perhaps my comment above on `JavaProtobufLibrary` will help to clarify this. I think my understanding is that only the concrete `UnpackedJars` and `UnpackedWhls` target types would actually need to implement this trait.

Reply:
Turned into #7111.